### ISSUE TITLE
Implement upcoming activities

### ISSUE THREAD

**Issue by rafaelklaessen** [WELTEN/dojo-ibl-mobile#17]

**Comment by Titogelo** [comment]
The purpose of the upcoming activities is to show only the activities that are due to soon (10 days) in all the runs that the user is participating in.

If no activities are due to soon an empty layout saying no activities are due soon.

**Comment by rafaelklaessen** [comment]
Is there a way to request those from the API directly? Or do I need to check all runs for activities that are due soon and then show them?

**Comment by Titogelo** [comment]
Once I was busy this method https://github.com/WELTEN/dojo-ibl/commit/bf683776544efdec6ba04e8f5d53b4fae6ce0e50#diff-c3058dac9b45314cffefcd7a8f9b11e2R80

but I didn't have the time to test it properly. Since you are now looking it up I will be back to it. Leave this implementation for now until I give you a sign to test the new method.

**Comment by Titogelo** [comment]
Now I remember why I stopped with. Let me try to explain it.

The functionality of retrieving the activities that are due soon is implemented. So a user in his/her profile could have a list of activities that need to be done soon. However, to access one of those activities from the profile, it is not trivial. We need to know to which **run** do we want to go, and that's the tricky part.

The problem is that the deadlines are set in the activity which is at generalItem level. This means all runs for a game will have the same deadline. To solve the issue, the deadlines for each activities need to set at the run level (individual deadlines per inquiry group). 

Similar to what we have for the status of the activities, that is manage at run time so that an activity in a game can be in a status in the different groups. Likewise, an activity in a game can have different deadlines in the different groups. **[this part still needs to be implemented]**

In conclusion, I need to invest quite some efforts to fully implement this thing. For the mobile app this functionality is also need it.
